feat: Object list item Component styles

sKudum opened the pull request:
## Related Issue
Closes #1165 
Closes #1512 


## Description
Introducing  Object list item component, one of the variants of the list as pere Fiori3.

## Screenshots

<img width="374" alt="2020-09-21_23-39-52" src="https://user-images.githubusercontent.com/53487468/93804324-c547cb80-fc63-11ea-98e6-9e583b35cc65.png">



- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-styles/wiki/PR-Review-Checklist

Betrozov commented on src/_listObject.scss:
```diff
@@ -0,0 +1,228 @@
+
+@import "./listDefinitions.scss";
```
plz remove `.scss` it is not needed

sKudum replied:
styles are not getting loaded sometimes when I am removing of .scss.

Betrozov commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,496 @@
+and to represent semantics for respective element append element-- \`neutral\`, \`positive\`, \`critical\`, \`negative\`
+and \`informative\` status.
+`,
+        tags: ['f3', 'a11y', 'theme', 'development']
```
is it correct tags?

sKudum replied:
included, as per other list variants

Betrozov commented:
<img width="632" alt="Screenshot 2020-09-22 at 11 22 28" src="https://user-images.githubusercontent.com/8318592/93859281-1ab4c480-fcc6-11ea-8ed5-2e25758323d9.png">
letters cut off

Betrozov commented:
plz fix travis

dimamarksman commented on src/_listObject.scss:
```diff
@@ -0,0 +1,228 @@
+      width: 40%;
+      flex: 2 2 10%;
+      text-align: right;
+      padding-left: $fd-list-object-with-padding-value-one !important;
+      vertical-align: $fd-list-object-with-vertical-align-top !important;
+      margin-right: $fd-list-object-with-margin-value-zero !important;
```
Can we avoid `!important` usage?

sKudum replied:
we have default styles at the list item level, whereas object list-style values need to override those values and ! import help here to achieve the required styles.

dimamarksman commented on src/_listObject.scss:
```diff
@@ -0,0 +1,228 @@
+    }
+  }
+
+  .fd-object-status {
```
Is it allowed to use external selectors for styling?
Should not we create our ourn selector for `fd-object-status`?

sKudum replied:
we need all the styles of fd-object-status, only a few values in it are changing in the object list, so instead of rewriting whole styles again. I reusing the class and overriding the required values. The same is done where ever it is applicable.

dimamarksman replied:
@InnaAtanasova could you please tell if such approach is acceptable here?

sKudum replied:
updated it as per Inna's commets now

dimamarksman reviewed (commented):
Just questions

sKudum commented:
> <img alt="Screenshot 2020-09-22 at 11 22 28" width="632" src="https://user-images.githubusercontent.com/8318592/93859281-1ab4c480-fcc6-11ea-8ed5-2e25758323d9.png">
> 
> letters cut off

fixed

sKudum commented:
> plz fix travis

Travis build will fail, because we had [WIP] in the title, which was added to mention that it has not completed internal review. and the second reason is it needs at least one approval from fundamental team members. When we click on Details it gives us detailed information on the failure.

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,798 @@
+export const selection = () => `
+<h4>Object List Item With Row Selection</h4>
+<div style="max-width: 450px">
+<ul class="fd-list fd-list--object fd-list--selection fd-list__row-selection" role="list">
```
Selection lists should have listbox role:
```suggestion
<ul class="fd-list fd-list--object fd-list--selection fd-list__row-selection" role="listbox">
```

sKudum replied:
updated

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,798 @@
+<h4>Object List Item With Row Selection</h4>
+<div style="max-width: 450px">
+<ul class="fd-list fd-list--object fd-list--selection fd-list__row-selection" role="list">
+  <li role="listitem" tabindex="-1" class="fd-list__item">
```
Selection list item should have option role:
```suggestion
  <li role="option" tabindex="0" class="fd-list__item">
```

sKudum replied:
updated

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,798 @@
+export const multiSelection = () => `
+<h4>Object List Item With Multi Selection</h4>
+<div style="max-width: 450px">
+<ul class="fd-list fd-list--object fd-list--selection" role="list">
```
```suggestion
<ul class="fd-list fd-list--object fd-list--selection" role="listbox">
```

sKudum replied:
updated

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,798 @@
+<h4>Object List Item With Multi Selection</h4>
+<div style="max-width: 450px">
+<ul class="fd-list fd-list--object fd-list--selection" role="list">
+  <li role="listitem" tabindex="-1" class="fd-list__item is-selected">
```
```suggestion
  <li role="option" aria-selected="true" tabindex="0" class="fd-list__item is-selected">
```

sKudum replied:
updated

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,798 @@
+   </div>
+   </div> 
+  </li>
+  <li role="listitem" tabindex="-1" class="fd-list__item">
```
```suggestion
  <li role="option" aria-selected="true" tabindex="0" class="fd-list__item">
```

sKudum replied:
updated role and tab index values, as it is not a selected row did not update aria-selected.

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,798 @@
+  </li>
+  <li role="listitem" tabindex="-1" class="fd-list__item">
+  <div class="fd-form-item fd-list__form-item">
+          <input type="checkbox" class="fd-checkbox" id="Ai4ez6111Z" checked="" aria-labelledby="O09lk1">
```
```suggestion
          <input type="checkbox" class="fd-checkbox" id="Ai4ez6111Z" checked="">
```

sKudum replied:
updated

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,798 @@
+<ul class="fd-list fd-list--object fd-list--selection" role="list">
+  <li role="listitem" tabindex="-1" class="fd-list__item is-selected">
+  <div class="fd-form-item fd-list__form-item">
+          <input type="checkbox" class="fd-checkbox" id="Ai4ez6111Z" checked="" aria-labelledby="O09lk1">
```
Use unique ids.
```suggestion
          <input type="checkbox" class="fd-checkbox" id="Ai4ez6111Y" checked="">
```

sKudum replied:
updated

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,798 @@
+  <li role="listitem" tabindex="-1" class="fd-list__item is-selected">
+  <div class="fd-form-item fd-list__form-item">
+          <input type="checkbox" class="fd-checkbox" id="Ai4ez6111Z" checked="" aria-labelledby="O09lk1">
+          <label tabindex="-1" class="fd-checkbox__label" for="Ai4ez6111Z"></label>
```
```suggestion
          <label aria-label="select/de-select option 1" tabindex="-1" class="fd-checkbox__label" for="Ai4ez6111Y"></label>
```

sKudum replied:
updated

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,798 @@
+  <li role="listitem" tabindex="-1" class="fd-list__item">
+  <div class="fd-form-item fd-list__form-item">
+          <input type="checkbox" class="fd-checkbox" id="Ai4ez6111Z" checked="" aria-labelledby="O09lk1">
+          <label tabindex="-1" class="fd-checkbox__label" for="Ai4ez6111Z"></label>
```
```suggestion
          <label aria-label="select/de-select option 2" tabindex="-1" class="fd-checkbox__label" for="Ai4ez6111Z"></label>
```

sKudum replied:
updated

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,798 @@
+};
+
+export const object = () => `
+<h4>Object List Item</h4>
```
```suggestion
<h4 id="objectListItemHeader" >Object List Item</h4>
```

sKudum replied:
updated

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,798 @@
+
+export const object = () => `
+<h4>Object List Item</h4>
+<ul class="fd-list fd-list--object" role="list">
```
Would be a good idea to label all the unordered lists by something like the header.
```suggestion
<ul class="fd-list fd-list--object" role="list" aria-labelledby="objectListItemHeader">
```

sKudum replied:
updated

prsdthkr reviewed (changes requested):
Some accessibility improvement:

jbadan commented on src/_listDefinitions.scss:
```diff
@@ -40,6 +40,27 @@ $fd-list-navigation-navigator-bar-right: inset -0.1875rem 0 0 0 var(--sapSelecte
 $fd-list-navigation-navigator-bar-left: inset 0.1875rem 0 0 0 var(--sapSelectedColor);
 $fd-list-navigation-indicator-width: 2.75rem !default;
 
+$fd-list-optional-text-font-family: var(--sapUiFontFamily);
```
Is this the right variable? I don't believe `sapUiFontFamily` is something that is a part of @sap-theming repo

https://github.com/SAP/theming-base-content/blob/master/content/Base/baseLib/sap_fiori_3/css_variables.css

Additionally, things like fontFamily should be set by the reset and should be unecessary

sKudum replied:
updated sapUiFontFamily, where ever usage of fd-reset is not causing issues updated it as well.

jbadan commented on src/_listDefinitions.scss:
```diff
@@ -40,6 +40,27 @@ $fd-list-navigation-navigator-bar-right: inset -0.1875rem 0 0 0 var(--sapSelecte
 $fd-list-navigation-indicator-width: 2.75rem !default;
 
+$fd-list-optional-text-font-family: var(--sapUiFontFamily);
+$fd-list-optional-text-font-size: var(--sapMFontMediumSize);
```
`sapMFontMediumSize` is not from the theming repo - https://github.com/SAP/theming-base-content/blob/master/content/Base/baseLib/sap_fiori_3/css_variables.css

sKudum replied:
updated

jbadan commented on src/_listDefinitions.scss:
```diff
@@ -40,6 +40,27 @@ $fd-list-navigation-navigator-bar-right: inset -0.1875rem 0 0 0 var(--sapSelecte
+$fd-list-optional-text-font-family: var(--sapUiFontFamily);
+$fd-list-optional-text-font-size: var(--sapMFontMediumSize);
+$fd-list-optional-text-font-weight: normal;
+$fd-list-optional-text-font-color: var(--sapUiContentLabelColor);
```
`sapUiContentLabelColor` is not the correct variable

sKudum replied:
updated

jbadan commented on src/_listDefinitions.scss:
```diff
@@ -40,6 +40,27 @@ $fd-list-navigation-navigator-bar-right: inset -0.1875rem 0 0 0 var(--sapSelecte
+$fd-object-identifier-title-font-family: var(--sapFontHeaderFamily);
+$fd-object-identifier-title-font-size: var(--sapFontHeader4Size);
+$fd-object-identifier-title-font-weight: var(--sapFontHeaderWeight);
+$fd-object-identifier-title-color: var(--sapGroup_TitleTextColor);
```
A lot of these variables seem unnecessary - can you do an audit that they are 1) the correct variables and 2) that they are actually needed and not already set by `fd-reset()`

sKudum replied:
where ever usage of fd-reset and fd-typography is not causing issues updated it as well and removed un required variables.

jbadan reviewed (changes requested):
See comments

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,900 @@
+<h4 id="objectListItemNavigation">Object List Item With Navigation</h4>
+
+<div style="max-width: 450px">
+<ul class="fd-list fd-list--object fd-list--navigation" role="listbox" aria-labelledby="object List Item With Navigation">
```
`aria-labelledby` value should be the `id` of the header above for all `<ul>`s
```suggestion
<ul class="fd-list fd-list--object fd-list--navigation" role="listbox" aria-labelledby="objectListItemNavigation">
```

sKudum replied:
updated

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,900 @@
+export const multiSelection = () => `
+<h4 id="objectListItemMultiSelection">Object List Item With Multi Selection</h4>
+<div style="max-width: 450px">
+<ul class="fd-list fd-list--object fd-list--selection" role="listbox" aria-labelledby="objectListItem with multi selection">
```
`aria-multiselectable="true"` should be added to multi selectable object list

```suggestion
<ul class="fd-list fd-list--object fd-list--selection" role="listbox" aria-multiselectable="true" aria-labelledby="objectListItemMultiSelection">
```

sKudum replied:
updated

prsdthkr reviewed (changes requested):
Some a11y changes.

prsdthkr commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,900 @@
+    title: 'Components/List/Object',
+    parameters: {
+        description: `
+        Object list provides a structure for Object list item content.
```
```suggestion
Object list provides a structure for Object list item content.
```

sKudum replied:
updated

prsdthkr reviewed (changes requested):
Styles are broken on IE 11.
![object-list-IE-styles](https://user-images.githubusercontent.com/43764832/94108100-8b0d3400-fdf3-11ea-852c-4a5c911cb1ff.gif)

InnaAtanasova commented on src/_listObject.scss:
```diff
@@ -0,0 +1,226 @@
+      }
+
+      &--top-left {
+        flex: 3 3 10%;
```
This will most probably cause problems in IE11

sKudum replied:
updated.

jbadan commented on src/_listObject.scss:
```diff
@@ -0,0 +1,226 @@
+      }
+    }
+
+    .fd-object-number {
```
I would prefer to avoid this pattern. We should add an additional class to `fd-object-number` and not use the same classes.

sKudum replied:
updated, as suggested by Inna in below comments.

sKudum commented:
> Styles are broken on IE 11.
> ![object-list-IE-styles](https://user-images.githubusercontent.com/43764832/94108100-8b0d3400-fdf3-11ea-852c-4a5c911cb1ff.gif)
updated the details whiich may cause issues in IE, as I don't have thr browser can't test the fix, please let me know in case issue still presists

sKudum commented on src/_listObject.scss:
```diff
@@ -0,0 +1,226 @@
+      min-width: $fd-list-object-with-padding-value-three;
+      max-width: $fd-list-object-with-padding-value-three;
+      min-height: $fd-list-object-with-padding-value-three;
+      max-height: $fd-list-object-with-padding-value-three;
```
updated

sKudum commented on src/_listObject.scss:
```diff
@@ -0,0 +1,226 @@
+      margin-right: 0.75rem;
+
+      @include fd-rtl() {
+        margin-left: 0.75rem;
```
updated

sKudum commented on src/_listObject.scss:
```diff
@@ -0,0 +1,226 @@
+      }
+    }
+
+    .fd-object-status {
```
updated

sKudum commented on src/_listObject.scss:
```diff
@@ -0,0 +1,226 @@
+      max-width: $fd-list-object-with-padding-value-three;
+      min-height: $fd-list-object-with-padding-value-three;
+      max-height: $fd-list-object-with-padding-value-three;
+      margin-right: 0.75rem;
```
updated

sKudum commented on src/_listObject.scss:
```diff
@@ -0,0 +1,226 @@
+      @include fd-reset();
+
+      width: 100%;
+      text-align: left;
```
updated

sKudum commented on src/_listObject.scss:
```diff
@@ -0,0 +1,226 @@
+
+      text-align: right;
+      width: 100%;
+      display: initial;
```
updated

sKudum commented on src/_listObject.scss:
```diff
@@ -0,0 +1,226 @@
+        -webkit-box-orient: vertical;
+        overflow: hidden;
+        white-space: normal;
+        text-overflow: ellipsis;
```
updated

sKudum commented on src/_listObject.scss:
```diff
@@ -0,0 +1,226 @@
+      &--bottom-left {
+        flex: 3 3 10%;
+        text-align: left;
+        padding: 0.5rem 0.5rem 0.5rem 0;
```
updated

sKudum commented on src/_listObject.scss:
```diff
@@ -0,0 +1,226 @@
+        flex: 3 3 10%;
+        text-align: left;
+        padding: 0.5rem 0.5rem 0.5rem 0;
+        display: inline-grid;
```
updated

sKudum replied:
As (parent classes)fd-list_item, _content has display:flex by which truncation was not happening, when I tried a different approach then display inline grid and initial helped to achieve it so I added them.
if you can share the suggested approach for truncation that will help. I will follow the same.
Thank you.

jbadan commented:
> > updated the details whiich may cause issues in IE, as I don't have thr browser can't test the fix, please let me know in case issue still presists

@sKudum How was it possible to complete the pr review checklist without having a working virtual machine with IE11? 
![Screen Shot 2020-09-25 at 8 33 19 AM](https://user-images.githubusercontent.com/29607818/94286551-006b2880-ff0a-11ea-94fc-739ecee1dcd4.png)


The checklist is not there by accident. You should download a windows machine from VirtualBox or vmware. Storybook is setup to be easily tested in that environment.

sKudum commented:
> > > updated the details whiich may cause issues in IE, as I don't have thr browser can't test the fix, please let me know in case issue still presists
> 
> @sKudum How was it possible to complete the pr review checklist without having a working virtual machine with IE11?
> ![Screen Shot 2020-09-25 at 8 33 19 AM](https://user-images.githubusercontent.com/29607818/94286551-006b2880-ff0a-11ea-94fc-739ecee1dcd4.png)
> 
> The checklist is not there by accident. You should download a windows machine from VirtualBox or VMware. The storybook is setup to be easily tested in that environment.

Hi @jbadan , I use Citrix in general to test, but since last 2 days am not able to open it due to network issues so (meanwhile the issue get resolved), I requested prasad in case he comes across the issue again I can look into it by trying other ways. 
because when I tested before raising the PR did not came across this issue.

jbadan commented on stories/list/object/object-list.stories.js:
```diff
@@ -0,0 +1,1003 @@
+import '../../../dist/checkbox.css';
+
+export default {
+    title: 'Components/List/Object',
```
@jessmetivier Should this be under "List" or grouped with the other "Object" components. This looks a little weird to me:

![Screen Shot 2020-09-25 at 9 59 52 AM](https://user-images.githubusercontent.com/29607818/94295303-e33c5700-ff15-11ea-9044-7994eaa92f92.png)

sKudum replied:
As per Fiori3 design guidelines, the object list item is one of the list variants and it will be reusing most the list and standard list item styles. 
Following the above points, it is placed under the list. Please let me know if that is not correct understanding, so I can place it accordingly.

jbadan commented on src/_listDefinitions.scss:
```diff
@@ -40,6 +40,22 @@ $fd-list-navigation-navigator-bar-right: inset -0.1875rem 0 0 0 var(--sapSelecte
+$fd-list-object-align-right: right;
+$fd-list-object-with-vertical-align-top: top;
+$fd-list-object-with-vertical-align-bottom: bottom;
+$fd-list-object-with-no-text-shadow: 0 0 black;
```
Is this correct? I can't find it in the spec. If it should be here, it should definitely be using a variable. It looks pretty strange:
With:
![Screen Shot 2020-09-25 at 10 01 09 AM](https://user-images.githubusercontent.com/29607818/94295492-2b5b7980-ff16-11ea-968e-31611de191d1.png)
![Screen Shot 2020-09-25 at 10 01 17 AM](https://user-images.githubusercontent.com/29607818/94295536-3adac280-ff16-11ea-937e-f9d6a1486edf.png)

sKudum replied:
Object variants such as status, number by default have a shadow effect on text when it is active but in the case of object lists item, these shadow effects should not be present as per the images shared in Fiori guidelines. To stay in sync with object list item styles look and feel this property is updated here.

As per Fiori.

<img width="518" alt="2020-09-26_00-30-00" src="https://user-images.githubusercontent.com/53487468/94305831-7b2a5700-ff8f-11ea-91cc-e66479ef233b.png">



Without text-shadow update look in local.

<img width="388" alt="2020-09-26_00-19-22" src="https://user-images.githubusercontent.com/53487468/94305903-97c68f00-ff8f-11ea-993c-ee14f944d0e5.png">

jbadan replied:
Can you show me where this is defined in the design spec? It's not mentioned in ObjectList or ObjectNumber:
![Screen Shot 2020-09-25 at 12 43 26 PM](https://user-images.githubusercontent.com/29607818/94309360-ceb78900-ff2c-11ea-9a50-38de2ae861ec.png)
![Screen Shot 2020-09-25 at 12 43 52 PM](https://user-images.githubusercontent.com/29607818/94309362-d1b27980-ff2c-11ea-90af-3696679d7b39.png)

sKudum replied:
Yes, that's right, inspite of this the current styles of object number and object status are showing shadow effect on text which in turn effecting object list item look as mentioned in the above images.

jbadan replied:
Sorry, I don't think I'm following. Why are you adding text shadow? Are you doing that based on the image you added with a blue background? If so, where is that coming from?

sKudum replied:
Object number and object status.scss has defined text-shadow (though it is not mentioned is the spec) which is resulting in updated text-shadow to match as per-object list item look as mentioned in the screenshot above.

jbadan replied:
I'm sorry, I'm still not following. How does it have a defined text-shadow if it's not mentioned in the spec? Are there incorrect styles in `object-status.scss` that should be removed instead of doing hacks to override them?

jbadan replied:
Doing some research, I see now where it's added. That contrast in objectStatus was added for accessibility reasons - https://github.com/SAP/fundamental-styles/pull/1571. We shouldn't need to override it here.

sKudum replied:
As per my understanding from the spec, yes we should not have text-shadow in object number and object status styles. 

As am not ware of the requirement behind adding the text-shadow there, I prefer to override it in object list item (As a temporary fix).

Until it is fixed (if it is considered as bug) in object number and object status.

jbadan replied:
The current text shadow was added as an accessibility fix. @jacobdevera can you provide additional context for the text shadow added in  https://github.com/SAP/fundamental-styles/pull/1571?

A possible override (if @jacobdevera can confirm that his changes were meant only for when the components are in a table and not in general), would be to set `text-shadow: none`.

jacobdevera replied:
The `text-shadow` was added primarily for Table when rows are selected as the selected color causes contrast issues. If it needs to be overridden, it makes sense to use `text-shadow: none`, as the black `text-shadow` currently being set seems off.

The reason it looks off in your screenshot is due to the inverted styling (white text against dark background) making it appear to have a glow effect. I also set the `text-shadow` to `none` in those cases as there doesn't seem to be an appropriate `text-shadow` variable to use for inverted text.

jacobdevera replied:
That said, the Object List does appear to have a selection mode with semantic text, which would run into the same problem as Table as it uses the same colors as Table. If it must be added to the spec to be incorporated here, the text-shadow should be added or be accepted as a known deviation, as the Positive text does _narrowly_ go below a 4.5:1 contrast without the text shadow.

jbadan commented on src/_listObject.scss:
```diff
@@ -0,0 +1,220 @@
+      }
+
+      &-unit {
+        font-family: $fd-list-object-number-font-family;
```
You're overriding the same value here coming from `fd-object-number__unit`. This should be removed:

![Screen Shot 2020-09-25 at 10 08 28 AM](https://user-images.githubusercontent.com/29607818/94296087-221edc80-ff17-11ea-9d30-168d478492ac.png)

sKudum replied:
updated

jbadan commented on src/_listObject.scss:
```diff
@@ -0,0 +1,220 @@
+
+      &-unit {
+        font-family: $fd-list-object-number-font-family;
+        font-size: $fd-list-object-font-size;
```
Same with font-size - it's already set to this. The design spec says it should be `1.375rem;`

sKudum replied:
The number needs a front-light as per fiori3, 
if we remove this it takes font family bold. that's, why we added, have this here.

jbadan commented on src/_listObject.scss:
```diff
@@ -0,0 +1,220 @@
+      &-unit {
+        font-family: $fd-list-object-number-font-family;
+        font-size: $fd-list-object-font-size;
+        text-shadow: $fd-list-object-with-no-text-shadow;
```
This isn't in the spec. Please remove

sKudum replied:
added too stay in object list item look as per Fiori3 specs.(as mentioned in the above comments.)

jbadan commented on src/_listObject.scss:
```diff
@@ -0,0 +1,220 @@
+        font-weight: normal;
+        font-size: 1.375rem;
+        font-family: $fd-list-object-number-font-family;
+        text-shadow: $fd-list-object-with-no-text-shadow;
```
Same as unit - the only thing that should stay in this block is the font-size.

sKudum replied:
text-shawdow -added to stay in object list item look as per Fiori3 specs.(as mentioned in the above comments.)

jbadan commented on src/_listObject.scss:
```diff
@@ -0,0 +1,220 @@
+
+    .#{$block}__object-identifier {
+      &-title {
+        @include fd-reset();
```
Why does this need `fd-reset`? It's not a new element block. Please remove

sKudum replied:
updated

jbadan commented on src/_listObject.scss:
```diff
@@ -0,0 +1,220 @@
+
+        line-height: 1.5;
+        vertical-align: $fd-list-object-with-vertical-align-top;
+        font-weight: $fd-object-identifier-title-font-weight;
```
Please remove font-weight, it's now set to the same value 4 times on this element.

sKudum replied:
updated

jbadan commented on src/_listObject.scss:
```diff
@@ -0,0 +1,220 @@
+      margin-right: $fd-list-object-with-value-zero;
+
+      &-text {
+        text-shadow: $fd-list-object-with-no-text-shadow;
```
Remove text-shadow

sKudum replied:
added to stay in object list item look as per Fiori3 specs.(as mentioned in the above comments.)

jbadan commented on src/_listObject.scss:
```diff
@@ -0,0 +1,220 @@
+
+    .#{$block}__object-attribute {
+      @include fd-ellipsis();
+      @include fd-reset();
```
Why is fd-ellipsis before fd-reset?

sKudum replied:
updated

jbadan reviewed (changes requested):
In addition to my comments about the css, something feels wrong about this whole component. Should it really be part of "List" and not it's own standalone "ObjectList" component?

sKudum commented:
Hi @jbadan, I have updated the changes as suggested but for a few such as text-shadow and font family, size removal is causing a mismatch with the object list item look as mentioned in Fiori.

 If you feel having a mismatch in look is ok or any other approach I need to follow, please do let me know.

 I will update them accordingly.

Thank You.

jbadan commented:
@sKudum I think this should be rebuilt as `fd-object-status` component. In my opinion, mixing it in with `fd-list` is confusing.  That will help this issue with following BEM naming conventions. Open to other opinions of course though! 😃

sKudum commented:
> @sKudum I think this should be rebuilt as `fd-object-status` component. In my opinion, mixing it in with `fd-list` is confusing. That will help this issue with following BEM naming conventions. Open to other opinions of course though! 😃

Object number and object status.scss has defined text-shadow (though it is not mentioned is the spec) which is resulting in updated text-shadow to match the per-object list item look as mentioned in the screenshot above.

I would like to take other reviewers' opinions on this, as they also suggested to not using component class names directly to override.

jbadan commented:
@sKudum I meant `fd-object-list`, it is correct that we shouldn't be using other component names in the other scss files.

InnaAtanasova commented:
Hi @sKudum , I share @jbadan opinion on having `fd-object-list` as a separate component. Right now we are violating BEM rules in multiple places. For example, `<div class="fd-list__object fd-list__object--top-row"></div>` is completely against BEM guidelines. The `fd-list__object--top-row` is not a modifier class but a container (a.k.a element) and should not be using `--`. The correct markup would be `<div class="fd-object-list fd-object-list__top-row"></div>` , indicating clearly that the top row is an element of `fd-object-list`.

Element D from the specs - Number: it's specified that this component inherits the styling and behaviour from the "Large Design" ObjectNumber (Fiori 3) with some deviation on the font-size and Element E is following the styling and behaviour from the ObjectNumber with no deviations. In this case you need to have a markup like this:
```
<span class="fd-object-number fd-object-number--large fd-object-number--critical fd-object-list__object-marker">
    <span class="fd-object-number__text">0.00</span><span class="fd-object-number__unit">EUR</span>
</span>
```
and then in css you only need to overwrite the font-size to be 1.375rem. You shouldn't be modifying the font-weight, the text-shadow or the font-family. For the unit (element E) you shouldn't have an additional class and overwrite anything. 

For element C - you need to follow the Avatar component. Avatar with size S has a width and height of 3rem so you don't need any additional classes on this element. The right spacing specified in the specs can be added as a left-spacing on the Title container and it can be removed if no avatar is added (simply checking if the title container is a first child and if it is, remove the left spacing). This way the user won't have to add an additional class to the Avatar just for the spacing and also it will be easier to handle the RTL.

InnaAtanasova commented:
For the class names I'd suggest a simplified naming. 
You currently have:
```
<div class="fd-list__object fd-list__object--bottom-row">
    <div class="fd-list__object fd-list__object--bottom-left">
        <div class="fd-list__object-attribute">
                  <span class="fd-list__object-attribute-text">First Attribute</span>
        </div>
    </div>
</div>
```

You can name each row in the component `fd-object-list__row` and either use a modifier classes like `fd-object-list__row--top` and `fd-object-list__row--bottom` or rely on the specificity and have `.fd-object-list__top .fd-object-list__row` and `.fd-object-list__bottom .fd-object-list__row`. 
Then your markup will be very simple:
```
<div class="fd-object-list__bottom">
     <div class="fd-object-list__row">
         <div class="fd-object-list__attribute">
                <span class="fd-object-list__attribute-text"> First Attribute</span>
         </div>
        <div class="fd-object-list__status">
                <span class="fd-object-status--inverted fd-object-status--critical">Inverted Warning</span>
        </div>
    </div>
</div>
```

InnaAtanasova commented:
You can also avoid names like `top` and `bottom` by using something like:
```
<li class="fd-object-list__item">
    <div class="fd-object-list__intro">...</div>
    <div class="fd-object-list__header">...</div>
    <div class="fd-object-list__content">...</div>
</li>
```

JKMarkowski commented on src/_listObject.scss:
```diff
@@ -0,0 +1,218 @@
+    }
+
+    .#{$block}__content {
+      flex: auto;
```
@include fd-reset(); missing

JKMarkowski commented on src/_listObject.scss:
```diff
@@ -0,0 +1,218 @@
+      &--top-right {
+        width: 40%;
+        text-align: $fd-list-object-align-right;
+        padding-left: $fd-list-object-with-padding-value-one !important;
+        vertical-align: $fd-list-object-with-vertical-align-top !important;
+        margin-right: $fd-list-object-with-value-zero !important;
```
Please remove `!important`

JKMarkowski commented on src/_listObject.scss:
```diff
@@ -0,0 +1,218 @@
+
+        @include fd-rtl() {
+          text-align: $fd-list-object-align-left;
+          padding-right: $fd-list-object-with-padding-value-one !important;
+          padding-left: $fd-list-object-with-value-zero !important;
+          margin-left: $fd-list-object-with-value-zero !important;
```
Let' remove important everywhere. It's forbidden

JKMarkowski commented on src/_listObject.scss:
```diff
@@ -0,0 +1,218 @@
+    }
+
+    .#{$block}__object {
+      width: 100%;
```
You should change it to `flex: 1 1 0`, also `fd-list__object-attribute` whouldn't have width: 50%;
Width with flex containers works quite poor, with `flex: 1 1 0` you would have:
- `100%` width, when there is only one object
- `50%` width (50/50) when there are two

JKMarkowski commented on src/_listObject.scss:
```diff
@@ -0,0 +1,218 @@
+        align-items: flex-end;
+        justify-content: flex-end;
+
+        > div:only-child {
```
That selector can't look like that, relying on pure elements (div) is forbidden

JKMarkowski reviewed (changes requested):
I added few comments regarding code, please take a look at Jenna's and Inna's sugestion, I totally agree with their comments. Let me know if it has been applied, so I will continue with review

sKudum commented:
> @sKudum I meant `fd-object-list`, it is correct that we shouldn't be using other component names in the other scss files.

Hi @jbadan, @InnaAtanasova , @JKMarkowski ,
Below are the changes done.

- Created object list as a new component.
- blocks, elements, modifier names are created as suggested.
- updated text-shadow to none.
- removed !important and used strong selectors instead of it.
- Width and reset details are updated as suggested.
-  Kept only relevant code as per the spec in the respective element classes such as object-identifier,object-attribute. 
- Only object number has display: block and content: none by which unit will be placed under number as shown in spec.
- For places I added comments, to understand why that piece of code is present will be removed once the review is done.

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+
+  &__header {
+    display: flex;
+    vertical-align: $fd-object-list-with-vertical-align-top;
```
This is used only in 1 place, no need to be a variable. We use variables if something is used in multiple places so in case something change in future it will be easier to update only one value

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+    vertical-align: $fd-object-list-with-vertical-align-top;
+
+    .#{$block}__header-left:not(:first-child) {
+      .#{$block}__object-identifier {
+        margin-left: $fd-object-list-with-value-half;
+
+        @include fd-rtl() {
+          margin-right: $fd-object-list-with-value-half;
+        }
+      }
```
This is not necessary. You can apply the margin-left on the container. You will have:
```
.#{$block}__header-left:not(:first-child) {
    margin-left: $fd-object-list-with-value-half;
    
    @include fd-rtl() {
           margin-left:0;
           margin-right: $fd-object-list-with-value-half;
    }
}
```

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+  }
+
+  &__header {
+    display: flex;
```
missing reset

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+    }
+  }
+
+  &__header-left {
```
missing reset

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+  }
+
+  &__header-right {
+    width: 40%;
```
missing reset

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+    // object number and object status extra margin values,
+    // by which all right items are not aligned same
+    //  comments will be removed after review
+    >[class*=number] {
```
not the way to select an element

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+    }
+  }
+
+  &__content {
```
missing reset

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+    }
+  }
+
+  &__row {
```
missing reset

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+    }
+  }
+
+  &__row-left {
```
missing reset

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+    }
+  }
+
+  &__row-right {
```
missing reset

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+      $fd-object-list-with-padding-value-half;
+    // added for object status as it parent has display:flex,
+    //  comments will be removed after review
+    >[class*=status] {
+      > * {
+        @include fd-ellipsis();
+      }
```
please re-write this

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+  }
+
+  &__object-attribute {
+    @include fd-reset();
+    @include fd-ellipsis();
+
+    &-text {
+      color: $fd-object-list-font-color;
+    }
+  }
```
You don't need a container that holds a simple text.

InnaAtanasova commented on src/object-list.scss:
```diff
@@ -0,0 +1,210 @@
+
+  &__object-identifier {
+    >[class*=identifier__title] {
+      @include fd-typography(var(--sapFontHeader4Size));
```
Please use just `font-size`

InnaAtanasova reviewed (changes requested):
Hi @sKudum , please change all selectors that are of the format  `>[class*=.....]` . This is not the way to target an element.

sKudum commented:
> Hi @sKudum , please change all selectors that are of the format `>[class*=.....]` . This is not the way to target an element.

Hi @InnaAtanasova , addressed all the comments .

- updated fd-reset added for requested elements.
- >[class*=.....] replaced with class names.
- used classname for span,tag,div ,i
- removed styles which are present in fd-list__item

jbadan commented on src/object-list.scss:
```diff
@@ -0,0 +1,212 @@
+        .#{$fd-namespace}-object-status__text,
+        .#{$fd-namespace}-object-status__icon {
+          color: var(--sapList_Active_TextColor);
+          text-shadow: none;
```
@jacobdevera Is it ok to remove the text shadow here? I thought we decided it should stay for a11y reasons.

sKudum replied:
Testers have raised concerns to handle text shadow that's why I made it as none.

 I removed it for now and will inform them to post here in case of concerns on this hope that will help.

jbadan reviewed (commented):
Pending a sign off from the accessibility team (@jacobdevera , @meganaconley , @Mike-Diaz ), I think these changes look good! Thanks for taking the time to redo this component. This iteration is more sustainable long term.

InnaAtanasova commented:
@sKudum just a question, not really a comment :) 
I see object list items have 1rem top and bottom paddings. In list with Byline, in borderless design, the "inner" items have 0.5rem top and bottom paddings, the first and last items have padding-bottom 0.5rem and padding-top 0.5rem respectively. I attach an image with what I'm trying to explain :D
<img width="893" alt="Screen Shot 2020-10-02 at 11 47 48 AM" src="https://user-images.githubusercontent.com/39598672/94943026-1c8c3e00-04a5-11eb-9dd7-de5426dbe7e2.png">

Do you know if this is also the case with Object list?

sKudum commented:
> @sKudum just a question, not really a comment :)
> I see object list items have 1rem top and bottom paddings. In list with Byline, in borderless design, the "inner" items have 0.5rem top and bottom paddings, the first and last items have padding-bottom 0.5rem and padding-top 0.5rem respectively. I attach an image with what I'm trying to explain :D
> <img alt="Screen Shot 2020-10-02 at 11 47 48 AM" width="893" src="https://user-images.githubusercontent.com/39598672/94943026-1c8c3e00-04a5-11eb-9dd7-de5426dbe7e2.png">
> 
> Do you know if this is also the case with Object list?

Not sure @InnaAtanasova on this did not come across this, as per the discussion with spec designers I came to know object list item still styles changes will happen as of now only PH1 ready. I remember the fundamentals team is also in that mail thread. if you want I can ask them again on this or else wait for some time as they suggested already.

InnaAtanasova commented:
The rest LGTM

InnaAtanasova commented:
> > @sKudum just a question, not really a comment :)
> > I see object list items have 1rem top and bottom paddings. In list with Byline, in borderless design, the "inner" items have 0.5rem top and bottom paddings, the first and last items have padding-bottom 0.5rem and padding-top 0.5rem respectively. I attach an image with what I'm trying to explain :D
> > <img alt="Screen Shot 2020-10-02 at 11 47 48 AM" width="893" src="https://user-images.githubusercontent.com/39598672/94943026-1c8c3e00-04a5-11eb-9dd7-de5426dbe7e2.png">
> > Do you know if this is also the case with Object list?
> 
> Not sure @InnaAtanasova on this did not come across this, as per the discussion with spec designers I came to know object list item still styles changes will happen as of now only PH1 ready. I remember the fundamentals team is also in that mail thread. if you want I can ask them again on this or else wait for some time as they suggested already.

@jessmetivier could you include this in the next email to the designers?
@sKudum it's not a blocker for the PR so you can merge it once it's approved, no need to wait for the designers answer. If something needs to be done it will be logged as a separate issue. Thanks for answering my question.

jacobdevera reviewed (approved):
LGTM 🚢

jbadan reviewed (approved):
🚀

prsdthkr commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+</div>
+`;
+
+selectionWithSelectionAndNavigation.storyName = 'selectionWithSelectionAndNavigation';
```
```suggestion
selectionWithSelectionAndNavigation.storyName = 'Selection with Selection and Navigation';
```

prsdthkr commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+
+export const object = () => `
+<h4 id="objectListItemHeader">Object List Item</h4>
+<ul class="fd-list fd-object-list" role="listbox" aria-labelledby="objectListItemHeader">
```
Only a selectable list should have a role `listbox`. Non-selectable lists should have the role `list`, which is the default.
```suggestion
<ul class="fd-list fd-object-list" aria-labelledby="objectListItemHeader">
```

prsdthkr commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+export const object = () => `
+<h4 id="objectListItemHeader">Object List Item</h4>
+<ul class="fd-list fd-object-list" role="listbox" aria-labelledby="objectListItemHeader">
+  <li role="option" tabindex="0" class="fd-list__item fd-object-list__item">
```
Only a selectable list item should have a role `option`. Non-selectable list item should have the role `listitem`, which is the default.
```suggestion
  <li tabindex="0" class="fd-list__item fd-object-list__item">
```

prsdthkr commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+<h4 id="objectListItemNavigation">Object List Item With Navigation</h4>
+
+<div style="max-width: 450px">
+<ul class="fd-list fd-object-list fd-list--navigation-object fd-list--navigation" role="listbox" aria-labelledby="objectListItemNavigation">
```
Navigation items need to be wrapped in the navigation role with a valid label. See https://www.w3.org/TR/wai-aria-practices/examples/landmarks/navigation.html 

```suggestion
<div role="navigation"  aria-labelledby="objectListItemNavigation">
<h4 id="objectListItemNavigation">Object List Item With Navigation</h4>
<ul class="fd-list fd-object-list fd-list--navigation-object fd-list--navigation">
```

prsdthkr commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+
+<div style="max-width: 450px">
+<ul class="fd-list fd-object-list fd-list--navigation-object fd-list--navigation" role="listbox" aria-labelledby="objectListItemNavigation">
+  <li role="option" tabindex="0" class="fd-list__item fd-object-list__item fd-list__item--link">
```
option role is only required in selectable list items.
```suggestion
  <li tabindex="0" class="fd-list__item fd-object-list__item fd-list__item--link">
```

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+selection.parameters = {
+    docs: {
+        iframeHeight: 305,
+        storyDescription: 'To create row selection list: Create a object list with `fd-list--selection` `fd-list--selection-row` modifiers.'
```
Object list items can be selectable by adding the `fd-list--selection` and `fd-list--selection-row` modifier classes to the main element.

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+
+selectionWithSelectionAndNavigation.storyName = 'selectionWithSelectionAndNavigation';
+
+selectionWithSelectionAndNavigation.parameters = {
```
Honestly, this is too convoluted... I think we can call this a "Complex" object list item.

sKudum replied:
renamed it to Selection with Navigation, few more features are expected in the next release of the version of the style from the Fiori team which have complex relations too. Let me know if this name does not suite.

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+selectionWithSelectionAndNavigation.parameters = {
+    docs: {
+        iframeHeight: 305,
+        storyDescription: ' To create row selection list with navigation add `fd-list--selection` `fd-list--navigation` `fd-list--navigation-object` modifier classes to the list and the `fd-list__item--link` modifier class to the list elements that contain links. All items should be navigable.'
```
The complex object list is both selectable and navigable. To display selectable object list items with navigation links, add the `fd-list--selection`, `fd-list--navigation` and `fd-list--navigation-object` modifier classes to the list and the `fd-list__item--link` modifier class to the list elements that contain links. All items should be navigable.

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+borderless.parameters = {
+    docs: {
+        iframeHeight: 305,
+        storyDescription: 'To create borderless list use `fd-list--no-border` modifier to list.'
```
Object list items can be displayed without borders. To display a borderless list, add the `fd-list--no-border` modifier class to the list element.

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+</div>
+`;
+
+rtl.storyName = 'Rtl';
```
Object list items can be displayed from right to left on the screen, so that they may be used internationally.

jessmetivier replied:
Capitalize RTL

jessmetivier reviewed (changes requested):
Well done! Please see my documentation updates, and let me know if you have any questions!

sKudum commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+</ul>
+`;
+
+object.storyName = 'Object';
```
> Modifiers are an overarching concept, so they should be mentioned with the main description.
> 
> What would you say about the examples here? Are they default object list items? How can we describe them?

No, these do not default, I placed them in main as suggested in another review.

sKudum replied:
No, these do not default, I placed them in main as suggested in another review.

jessmetivier replied:
How about 'Standard'? Would this be the basic way to style this component?

sKudum commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+- Object Status
+
+## Structure
+**Object List consists of the following elements along with standard list:**
```
I indented to say, in the below structure of object list item classes such as fd-list__item etc are used the same as standard list item do. 
But now I feel it may not be required to mention because it is common to all list variants.

sKudum commented:
Hi @jessmetivier , @prsdthkr addressed your comments, please do let me know in case anything needs to be done further.

prsdthkr commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1194 @@
+<h4 id="objectListItemRowSelection">Object List Item With Row Selection</h4>
+<div style="max-width: 450px">
+<ul class="fd-list fd-object-list fd-list--selection fd-list--selection-row" role="listbox" aria-labelledby="objectListItemRowSelection">
+  <li role="option" tabindex="0" class="fd-list__item fd-object-list__item is-selected">
```
```suggestion
  <li role="option" aria-selected="true" tabindex="0" class="fd-list__item fd-object-list__item is-selected">
```

prsdthkr commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1194 @@
+    </div>
+    </div>
+  </li>
+  <li role="option" tabindex="0" class="fd-list__item fd-object-list__item">
```
```suggestion
  <li role="option" aria-selected="false" tabindex="0" class="fd-list__item fd-object-list__item">
```

prsdthkr commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1194 @@
+<h4 id="objectListItemRowSelectionAndNavigation">Object List Item With Row Selection And Navigation</h4>
+<div role="navigation" style="max-width: 450px">
+<ul class="fd-list fd-object-list fd-list--navigation-object fd-list--navigation fd-list--selection" role="listbox" aria-labelledby="objectListItemRowSelectionAndNavigation">
+  <li role="option" tabindex="0" class="fd-list__item fd-object-list__item fd-list__item--link is-selected">
```
```suggestion
  <li role="option" aria-selected="true" tabindex="0" class="fd-list__item fd-object-list__item fd-list__item--link is-selected">
```

prsdthkr commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1194 @@
+    </div>
+    </a>
+  </li>
+  <li role="option" tabindex="0" class="fd-list__item fd-object-list__item fd-list__item--link">
```
```suggestion
  <li role="option" aria-selected="false" tabindex="0" class="fd-list__item fd-object-list__item fd-list__item--link">
```

prsdthkr reviewed (changes requested):
Sorry did not notice this earlier. Here are some final suggestions for accessibility.

prsdthkr reviewed (approved):
LGTM

JKMarkowski reviewed (approved):
LGTM

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1194 @@
+- Title of the object
+- Avatar
+- Object display components such as **Object Identifier**, **Object Number**, **Object Attribute**, **Object Marker** and **Object Status**
+- Introductory text indicating the origin of the object, such as -Forwarded by...- or -On Behalf of...-
```
Oops, I think I meant to put underscores for "-Forwarded by...- or -On Behalf of...-". They should be italicized like this: _Forwarded by..._ or _On Behalf of..._

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1194 @@
+
+
+## Structure
+**Object List consists of the following elements:**
```
Decapitalize the first letter in "List", so it should be "Object list"

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1194 @@
+  flagged for follow-up
+
+##Usage
+**Use the object List if:**
```
Decapitalize L in "List"

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1194 @@
+
+- You want to display a brief details to user.
+
+**Do not use the object List if:**
```
Decapitalize L in "List"

jessmetivier reviewed (changes requested):
This looks great! Just a few minor adjustments, and need to establish what the first component description is.

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1194 @@
+</ul>
+`;
+
+object.storyName = 'Object';
```
**Standard**
The standard object list item displays a title, introductory text, an avatar and object display components. It is recommended to display avatar in size S, by adding the `fd-avatar--s` modifier class to the `fd-avatar` element.

jessmetivier reviewed (commented):
Hi Sushma,

I've added a description for the standard object list item! Just title it as "Standard", like we've done in other docs :)

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+    title: 'Components/Object List',
+    parameters: {
+        description: `
+An object list is another type of list that consists of object list items, and each item provides a quick 
+overview of an object. Object list items should only contain essential information for the user to identify what 
+objects to prioritize. Avoid long descriptive texts as the text space is limited and will truncate.
```
Just a slight correction of my previous intro:

An object list is a type of list that consists of object list items, where each item provides a quick 
overview of an object. Object list items should only contain essential information for the user to identify what objects to prioritize. Avoid long descriptive texts as the text space is limited and will truncate.

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+  flagged for follow-up
+
+##Usage
+**Use the object list if:**
+
+- You want to display a brief details to user.
+
+**Do not use the object list if:**
+
+- You want to display a detailed information to user.
```
**Use the object list if:**

- You want to display brief details to the user.

**Do not use the object list if:**

- You want to display detailed information to the user.
- You want to trigger more than one event. Instead, consider using **Action List Item**.

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+selectionWithNavigation.parameters = {
+    docs: {
+        iframeHeight: 305,
+        storyDescription: 'This object list is both selectable and navigable. To display selectable object list items with navigation links, add the `fd-list--selection`, `fd-list--navigation` and `fd-list--navigation-object` modifier classes to the list and the `fd-list__item--link` modifier class to the list elements that contain links. All items should be navigable.'
```
Let's change this to:

This object list is both selectable and navigable. To display selectable object list items with navigation, add the `fd-list--selection`, `fd-list--navigation` and `fd-list--navigation-object` modifier classes to the list and the `fd-list__item--link` modifier class to the list elements that contain links. All items should be navigable.

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+borderless.parameters = {
+    docs: {
+        iframeHeight: 305,
+        storyDescription: 'Object list items can be displayed without borders. To display a borderless list, add the `fd-list--no-border` modifier class to the list element'
```
Missing period at the end of the sentence

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+object.parameters = {
+    docs: {
+        iframeHeight: 785,
+        storyDescription: 'The standard object list item displays a title, introductory text, an avatar and object display components. It is recommended to display avatar in size S, by adding the `fd-avatar--s` modifier class to the `fd-avatar` element.'
```
Let's change this to:

The standard object list item displays a title, introductory text, an avatar and object display components. It is recommended to display avatar in size S, therefore add the `fd-avatar--s` modifier class to the `fd-avatar` element.

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1193 @@
+navigation.parameters = {
+    docs: {
+        iframeHeight: 355,
+        storyDescription: 'An object list can display links that the users can navigate to. To display a navigation object list, add the `fd-list--navigation-object` modifier class together with `fd-list--navigation` and the `fd-list__item--link` modifier class to the list elements that contain links. All items should be navigable.'
```
Let's change this to:

An object list can display links that the users can navigate to. To display a navigation object list, add the `fd-list--navigation-object` modifier class together with `fd-list--navigation` and the `fd-list__item--link` modifier class to the list elements that contain a link. All items should be navigable.

jessmetivier reviewed (changes requested):
This should be the last one ;)

jessmetivier commented on stories/object-list/object-list.stories.js:
```diff
@@ -0,0 +1,1194 @@
+**Do not use the object list if:**
+
+- You want to display detailed information to the user.
+- You want to trigger more than one event. Instead, consider using Action List Item.
```
Oh right, we don't have Action List Item... so this should instead be:

You want to trigger more than one event, as each object list item should contain one link.

jessmetivier reviewed (changes requested):
Hi Sushma,

I've added two comments, please address :)

I'll approve anyway, because they are so minor and I trust you're on the ball with making the changes!
